### PR TITLE
refactor(samples): convert sample tests from ava to mocha

### PR DESCRIPTION
**Pull request by nareshqlogic**
Fixes convert all sample tests from ava to mocha [#2865](https://github.com/googleapis/google-cloud-node/issues/2865) (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/package.json [hunk 1]
```diff
@@ -9,13 +9,10 @@
...
   "scripts": {
-    "ava": "ava -T 20s --verbose test/*.test.js system-test/*.test.js",
-    "cover": "nyc --reporter=lcov --cache ava -T 20s --verbose test/*.test.js system-test/*.test.js && nyc report",
     "error-test": "samples test app --msg \"Something broke!\" --url \"http://localhost:33332/error\" --port 33332  -- snippets.js express",
     "exception-test": "samples test app --code 500 --msg SyntaxError --url \"http://localhost:33333/exception\" --port 33333  -- snippets.js express",
-    "system-test": "ava -T 1m --verbose system-test/*.test.js",
-    "all-test": "npm run system-test && npm run error-test && npm run exception-test",
-    "test": "npm run cover"
+    "all-test": "npm run test && npm run error-test && npm run exception-test",
+    "test": "mocha system-test/*.test.js --timeout=600000"
```
Comment (JustinBeckwith) [inline samples/package.json]: Looks like we're missing `mocha` in the samples here, and have `ava` still installed below

Comment (nareshqlogic) [reply samples/package.json]: Removed ava from devDependencies. Let me know if need to add mocha in devDependencies

Comment (nareshqlogic) [reply samples/package.json]: Added mocha in devDependencies

**Comment by nareshqlogic** [comment]
Hi @JustinBeckwith ,
System test are independent of this PR , right ?

**Pull request merged**